iOS/Controllers/HomeworkList/HomeworkSubmitViewController.swift

```diff
@@ -0,0 +1,363 @@
+        let discardAction = UIAlertAction(title: "Discard Changes", style: .destructive) { [unowned self] _ in
+            self.writingContext.rollback()
+            self.updateState()
+        }
+
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        [cancelAction, discardAction].forEach { alertController.addAction($0) }
+
+        self.present(alertController, animated: true)
+    }
+
+    @IBAction func submitFile(_ sender: Any) {
+        let picker = UIImagePickerController()
+        picker.delegate = self
+
+        let actionController = UIAlertController()
+        let cameraAction = UIAlertAction(title: "Taking a picture", style: .default) { [unowned self] _ in
+                picker.sourceType = .camera
+            picker.allowsEditing = true
+            picker.cameraCaptureMode = .photo
+            picker.cameraDevice = .rear
+            picker.showsCameraControls = true
+            self.present(picker, animated: true)
+        }
+
+        let libraryAction = UIAlertAction(title: "Photo Library", style: .default) { [unowned self] _ in
+            picker.sourceType = .savedPhotosAlbum
+            picker.allowsEditing = false
+
+            self.present(picker, animated: true)
+        }
+
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        [cameraAction, libraryAction, cancelAction].forEach { actionController.addAction($0) }
+
+        self.present(actionController, animated: true)
+    }
+
+    fileprivate func updateState() {
+        var fileIDs = Set<String>(self.submission.files.map { $0.id })
+        fileIDs.formUnion(self.writableSubmission.files.map { $0.id })
+        self.files = [String](fileIDs).sorted()
+
+        self.commentField.text = self.writableSubmission.comment ?? placeholder
+        self.filesTableView.reloadData()
+        self.tableViewHeightConstraint.constant = self.filesTableView.contentSize.height
+        self.contentView.setNeedsLayout()
+    }
+
+    func show(error: Error) {
+        let alertController = UIAlertController(title: "Something unexpected occured", message: error.localizedDescription, preferredStyle: .alert)
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+
+        self.present(alertController, animated: true)
+    }
+}
+
+extension HomeworkSubmitViewController: UITableViewDelegate {
+
+}
+
+extension HomeworkSubmitViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.files.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let fileId = self.files[indexPath.row]
+        if let cell = tableView.dequeueReusableCell(withIdentifier: "fileCell") {
```

Comment:
use `dequeueReusableCell(withIdentifier:for: )` to avoid the optional cell

https://developer.apple.com/documentation/uikit/uitableview/1614878-dequeuereusablecell